xcode/build_phases/swiftlint.rb

```diff
@@ -0,0 +1,393 @@
+            opt.on('-d', '--source_date DATE', 'The date of grouping files according touchin and old swiftlint rules') { |option| @options.source_date = option }
+            opt.on('-y', '--touchin_swiftlint_yaml_path STRING', 'The path to the touchin swiftlint yaml relative to the source directory') { |option| @options.touchin_swiftlint_yaml_path = option }
+            opt.on('-g', '--depth_git_count Int', 'The depth between the git directory and sources directory') { |option| @options.depth_git_count = option }
+        end.parse!
+        
+        if @options.check_mode.to_s.nilOrEmpty?
+            @options.check_mode = 'fully'
+        end
+
+        if @options.use_multiple.to_s.nilOrEmpty?
+            @options.use_multiple = 'false'
+        end
+
+        if @options.depth_git_count.to_s.nilOrEmpty?
+            @options.depth_git_count = 0
+        end
+        
+        if @options.touchin_swiftlint_yaml_path.to_s.nilOrEmpty?
+            @options.touchin_swiftlint_yaml_path = '/build-scripts/xcode/.swiftlint.yml'
+        end
+    end
+    
+    def source_directory
+        @options.source_directory
+    end
+    
+    def source_date
+        @options.source_date
+    end
+    
+    def pods_directory
+        @options.pods_directory
+    end
+    
+    def check_mode
+        @options.check_mode
+    end
+    
+    def use_multiple
+        @options.use_multiple
+    end
+    
+    def depth_git_count
+        @options.depth_git_count
+    end
+    
+    def touchin_swiftlint_yaml_path
+        @options.touchin_swiftlint_yaml_path
+    end
+end
+
+class CommandUtils
+    def self.make_command(command)
+        command = command.to_s
+        return `#{command}`
+    end
+end
+
+class GitСaretaker < CommandUtils
+    def self.get_modified_files
+        non_indexed_files = get_files_from('git diff --name-only | sed s/.*/"&,"/ ')
+        indexed_files = get_files_from('git diff --cached --name-only | sed s/.*/"&,"/ ')
+        
+        modified_files = non_indexed_files + indexed_files
+        unique_modified_files = modified_files.uniq
+        
+        unique_modified_swift_files = []
+        if not unique_modified_files.nilOrEmpty?
+            unique_modified_swift_files = unique_modified_files.select { |file_path|
+                file_path.to_s.filter_allowed_symbol_into_path
+                file_path.to_s.include? '.swift'
+            }
+        end
+        
+        return unique_modified_swift_files
+    end
+    
+    def self.get_creation_date(file_path)
+        git_command = 'git log --follow --format=%cD --reverse -- ' + file_path + ' | head -1'
+        return make_command(git_command)
+    end
+    
+    private
+    
+    def self.get_files_from(command)
+        files_as_string = make_command(command)
+        return files_as_string.split(',')
+    end
+end
+
+class SwiftFileManager
```

Comment:
to separate file

Reply:
without ruby ide is so hard watch 7 files on one monitor